src/wasm.rs

```diff
@@ -2,57 +2,56 @@
 
 use crate::{
     errors::{self, ErrorKind},
-    jwt::{Actor, Claims, Token},
+    jwt::{Actor, Claims, Token, MIN_WASCAP_INTERNAL_REVISION},
     Result,
 };
 use data_encoding::HEXUPPER;
 use nkeys::KeyPair;
-use parity_wasm::{
-    deserialize_buffer,
-    elements::{CustomSection, Module, Serialize},
-    serialize,
-};
 use ring::digest::{Context, Digest, SHA256};
 use std::{
-    io::Read,
+    io::{Read, Write},
     time::{SystemTime, UNIX_EPOCH},
 };
+use wasmparser::BinaryReaderError;
 const SECS_PER_DAY: u64 = 86400;
+const SECTION_JWT: &str = "jwt";
+const SECTION_WC_JWT: &str = "wasmcloud_jwt";
 
 /// Extracts a set of claims from the raw bytes of a WebAssembly module. In the case where no
 /// JWT is discovered in the module, this function returns `None`.
 /// If there is a token in the file with a valid hash, then you will get a `Token` back
 /// containing both the raw JWT and the decoded claims.
 ///
 /// # Errors
-/// Will return errors if the file cannot be read, cannot be parsed, contains an improperly
-/// forms JWT, or the `module_hash` claim inside the decoded JWT does not match the hash
-/// of the file.
+/// Will return an error if hash computation fails or it can't read the JWT from inside
+/// a section's data, etc
 pub fn extract_claims(contents: impl AsRef<[u8]>) -> Result<Option<Token<Actor>>> {
-    let module: Module = deserialize_buffer(contents.as_ref())?;
-
-    let sections: Vec<&CustomSection> = module
-        .custom_sections()
-        .filter(|sect| sect.name() == "jwt")
-        .collect();
-
-    if sections.is_empty() {
-        Ok(None)
-    } else {
-        let jwt = String::from_utf8(sections[0].payload().to_vec())?;
-        let claims: Claims<Actor> = Claims::decode(&jwt)?;
-        let hash = compute_hash_without_jwt(module)?;
-
-        if let Some(ref meta) = claims.metadata {
-            if meta.module_hash != hash {
-                Err(errors::new(ErrorKind::InvalidModuleHash))
-            } else {
-                Ok(Some(Token { jwt, claims }))
+    let parser = wasmparser::Parser::new(0);
+    for payload in parser.parse_all(contents.as_ref()) {
+        match payload? {
+            wasmparser::Payload::CustomSection(reader) => {
+                if reader.name() == SECTION_JWT || reader.name() == SECTION_WC_JWT {
```

Comment:
What's the reasoning for supporting a `jwt` and `wasmcloud_jwt` section? Was it `jwt` before and going forward we're using `wasmcloud_jwt`?